cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/ReportGenerationSpark.java

```diff
@@ -118,6 +118,60 @@
   protected void configure() {
     setMainClass(SparkPersistRunRecordMain.class);
     addHandlers(new ReportSparkHandler());
+    String version = getSparkVersion();
+    if (!isSparkVersionSupported(version)) {
+      throw new RuntimeException(
+        String.format("Spark Version %s is not supported, Please upgrade to Spark version 2.1 or later", version));
+    }
+  }
+
+  /**
+   * Spark version returned using SparkContext#version() is set by org.apache.spark.SparkBuildInfo in a similar manner
+   * by loading the spark-version-info.properties file and getting the property version.
+   * @return version string, if unable to load the file due to any error, return null
+   */
+  @Nullable
+  private String getSparkVersion() {
+    ClassLoader classLoader = org.apache.spark.SparkContext.class.getClassLoader();
+    InputStream inputStream = classLoader.getResourceAsStream("spark-version-info.properties");
```

Comment:
use try with resources so that inputstream is closed as well.
```try(InputStream inputStream = classLoader.getResourceAsStream("spark-version-info.properties")) {

} catch (IOException e) {
      return null;
   }```